src/Venflow/Venflow/Modeling/DatabaseConfigurationFactory.cs

```diff
@@ -112,19 +117,19 @@ private void CreateEntityConfigurations(Type databaseType, List<PropertyInfo> da
                 var entityType = property.PropertyType.GetGenericArguments()[0];
 
                 TypeFactory.AddEntityAssembly(entityType.Assembly.GetName().Name!);
+                var entityBuilderType = genericEntityBuilderType.MakeGenericType(entityType);
+                // var constructor = entityBuilderType.GetConstructor(new [] {typeof(Database), typeof(string)});
+                var entityBuilderInstance = Activator.CreateInstance(entityBuilderType, BindingFlags.NonPublic | BindingFlags.Instance, null, new object[] {configurationOptionsBuilder.EffectiveDatabase, property.Name}, null);
```

Comment:
I'd cast the `entityBuilderInstance` right here to an `EntityBuilder`, since we partially know the type here.

Reply:
done